File: 560-subarray-sum-equals-k/subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        Map<Integer,Integer> map = new HashMap<>();
        int prefix = 0;
        int ans = 0;
        map.put(0,1);
        for(int i=0;i<nums.length;i++){
            prefix+=nums[i];
            int diff = prefix-k;
            if(map.containsKey(diff)){
                ans+=map.get(diff);
            }
            map.put(prefix,map.getOrDefault(prefix,0)+1);
        }
        return ans;
    }
}

File: 560-subarray-sum-equals-k/subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        pre = 0
        ans=0
        map={}
        map[0] = 1
        for i in range (len(nums)):
            pre+=nums[i]
            target=pre-k
            if(target in map):
                ans+=map[target]
            map[pre] = map.get(pre,0)+1
        return ans
            
        